packages/xpetra/sup/Utils/Xpetra_MatrixUtils.hpp

```diff
@@ -453,7 +453,7 @@ class MatrixUtils {
     Ac->getLocalDiagCopy(*diagVec);
 
     LocalOrdinal lZeroDiags = 0;
-    Teuchos::ArrayRCP< Scalar > diagVal = diagVec->getDataNonConst(0);
+    Teuchos::ArrayRCP< const Scalar > diagVal = diagVec->getData(0);
```

Comment:
Huh, that's interesting.

Reply:
The diagonal is modified by building a new matrix, not by modifying this view.